crates/aiken-lang/src/tests/format.rs

```diff
@@ -560,3 +560,50 @@ fn test_unicode() {
 
     assert_fmt(src, src);
 }
+
+#[test]
+fn test_preserve_pipe() {
+    let src = indoc! { r#"
+        fn foo() {
+          a |> b |> c |> d
+        }
+
+        fn foo() {
+          a
+            // Foo
+            |> b// Some comments
+            |> c
+            |> d
+        }
+
+        fn baz() {
+          // Commented
+          however |> it_automatically_breaks |> into_multiple_lines |> anytime_when |> it_is_too_long // What?
+        }
+    "#};
+
+    let expected = indoc! { r#"
+        fn foo() {
+          a |> b |> c |> d
+        }
+
+        fn foo() {
+          a // Foo
+            |> b // Some comments
+            |> c
+            |> d
+        }
+
+        fn baz() {
+          // Commented
+          however
+            |> it_automatically_breaks
+            |> into_multiple_lines
+            |> anytime_when
+            |> it_is_too_long
```

Comment:
Note @rvcas, happy to discuss that one but I also introduced a level of indentation for pipelines. I find it to read a lot better than: 

```gleam
however
|> it_automatically_breaks
|> into_multiple_lines
|> anytime_when
|> it_is_too_long
```

Reply:
No problem, looks great